compiler/noirc_frontend/src/hir_def/types.rs

```diff
@@ -1940,7 +2067,7 @@ impl std::fmt::Debug for Type {
             }
             Type::Unit => write!(f, "()"),
             Type::Error => write!(f, "error"),
-            Type::NamedGeneric(binding, name) => write!(f, "{}{:?}", name, binding),
+            Type::NamedGeneric(binding, name, kind) => write!(f, "{} {}{:?}", kind, name, binding),
```

Comment:
Can you add a check to not display the kind if it is Normal? I think it'd help unclutter the output a bit.

I'm imagining we can display the existing `A123 -> <binding>` for `*` kinds and maybe something like `(A123 : i32) -> <binding>` for numeric named generics. The parenthesis and having the kind on the right would help distinguish it from other types if we ever do include the specific type like `i32` in the kind instead of just `numeric`.

Reply:
As I included the actual type as part of the numeric kind now I went with your suggestion of `(A123 : i32) -> <binding>` and `A123 -> <binding>` for `*` kinds.